scip-java/src/main/scala/com/sourcegraph/scip_java/ScipSymbol.scala

```diff
@@ -0,0 +1,55 @@
+package com.sourcegraph.scip_java
+
+import com.sourcegraph.scip_semanticdb.SymbolDescriptor
+import com.sourcegraph.semanticdb_javac.SemanticdbSymbols
+
+sealed abstract class ScipSymbol {}
+final case class LocalScipSymbol(identifier: String) extends ScipSymbol
+final case class GlobalScipSymbol(
+    scheme: String,
+    packageManager: String,
+    packageName: String,
+    packageVersion: String,
+    descriptors: List[SymbolDescriptor]
+) extends ScipSymbol
+
+object ScipSymbol {
+
+  def parseOrThrowExceptionIfInvalid(scipSymbol: String): ScipSymbol = {
+    if (scipSymbol.startsWith("local ")) {
+      LocalScipSymbol(scipSymbol.stripPrefix("local "))
+    } else {
+      scipSymbol.split(" ", 5) match {
+        case Array(
+              scheme,
+              packageManager,
+              packageName,
+              packageVersion,
+              descriptor
+            ) =>
+          GlobalScipSymbol(
+            scheme,
+            packageManager,
+            packageName,
+            packageVersion,
+            parseDescriptors(descriptor)
+          )
```

Comment:
If we are targeting 2.13, I think you can use direct matching on a string here?

```suggestion
      scipSymbol match {
        case s"$scheme $packageManager $packageName $packageVersion $descriptor" => 
          GlobalScipSymbol(
            scheme,
            packageManager,
            packageName,
            packageVersion,
            parseDescriptors(descriptor)
          )
```

I would check but my local is broken again. It's ridiculous.